examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx

```diff
@@ -41,21 +41,10 @@ export function AppointmentActions(props: AppointmentActionsProps): JSX.Element
   // Handler for completing or cancelling the appointment
   async function handleChangeStatus(newStatus: Appointment['status']): Promise<void> {
     try {
-      // TODO: Move code to a bot. Use patchResource instead of updateResource.
       await medplum.updateResource({
         ...appointment,
         status: newStatus,
-        slot: undefined,
       });
-      // If the appointment is cancelled, update the slot status to free
-      const slotId = resolveId(appointment.slot?.[0]);
-      if (newStatus === 'cancelled' && slotId) {
-        const slot = await medplum.readResource('Slot', slotId);
-        await medplum.updateResource({
-          ...slot,
-          status: 'free',
-        });
-      }
```

Comment:
Simplify logic. It's best to keep the slot tied to the appointment for consistency, even if the appointment has been canceled.